.github/workflows/ci.yml

```diff
@@ -53,24 +60,25 @@ jobs:
       - name: Install dependencies
         run: composer --verbose install
 
-      - if: matrix.drupal-release == 'dev'
-        run: composer --verbose require --no-update drupal/core-recommended:10.0.x-dev && composer --verbose require --no-update --dev drupal/core-dev:10.0.x-dev
-
-      - if: matrix.drupal-release == 'dev'
-        run: composer --verbose update
+      # @see https://www.drupal.org/node/3176567
+      - name: Install testing dependency
+        if: ${{ matrix.php-versions == '8.1' }}
+        run: composer require --dev phpspec/prophecy-phpunit:^2
 
-      - run: ./vendor/bin/drush site-install --verbose --yes --db-url=sqlite://tmp/site.sqlite
+      - name: Override Drupal version to dev for testing dev releases
+        if: matrix.drupal-release == 'dev'
+        run: |
+          composer --verbose require --no-update drupal/core-recommended:10.0.x-dev
+          composer --verbose require --no-update --dev drupal/core-dev:10.0.x-dev
+          composer --verbose update
 
-      - run: ./vendor/bin/drush runserver $SIMPLETEST_BASE_URL &
+      - name: Install site
+        run: ./vendor/bin/drush site-install --verbose --yes --db-url=sqlite://tmp/site.sqlite
 
-      - run: until curl -s $SIMPLETEST_BASE_URL; do true; done > /dev/null
+      - name: Start server
+        run: |
+          ./vendor/bin/drush runserver "$SIMPLETEST_BASE_URL" &
+          until curl -s "$SIMPLETEST_BASE_URL"; do true; done > /dev/null
 
-        # https://www.drupal.org/project/drupal/issues/3182653
-        # https://www.drupal.org/node/3176567
-      - if: ${{ matrix.php-versions == '8.1' }}
-        run: composer require phpspec/prophecy-phpunit:^2
-
-      - name: Run a single unit test to verify the testing setup.
+      - name: Run a single unit test to verify the testing setup
         run: ./vendor/bin/phpunit -c ./web/core ./web/core/modules/user/tests/src/Unit/UserAccessControlHandlerTest.php
-
-      - run: ./vendor/bin/drush
```

Comment:
this is not needed.- Drush was used above to install the site and run the server